packages/test-framework/src/fixture/pageObject/Checkout.ts

```diff
@@ -92,4 +93,143 @@ export class Checkout {
+            .fill(`${address.address1} ${addressAppendText}`);
+        await this.page
+            .locator(`${formId} [data-test="addressLine2Input-text"]`)
+            .fill(address.address2);
+        await this.page.locator(`${formId} [data-test="cityInput-text"]`).fill(address.city);
+        await this.page
+            .locator(`${formId} [data-test="countryCodeInput-select"]`)
+            .selectOption(address.countryCode);
+        await this.page
+            .locator(`${formId} [data-test="provinceCodeInput-select"]`)
+            .selectOption(address.stateOrProvinceCode);
+        await this.page
+            .locator(`${formId} [data-test="postCodeInput-text"]`)
+            .fill(address.postalCode);
+    }
+
+    async completeShippingAddressForm(
+        address: Address,
+        isBillingAddressSame = true,
+        shouldSaveAddress = true,
+        addressAppendText?: string,
+    ): Promise<void> {
+        await this.page
+            .locator('[data-test="shipping-address-heading"]')
+            .waitFor({ state: 'visible' });
+
+        await this.fillAddressForm({
+            formId: '#checkoutShippingAddress',
+            address,
+            addressAppendText,
+        });
+
+        if (!isBillingAddressSame) await this.page.locator('label[for="sameAsBilling"]').click();
+        if (!shouldSaveAddress)
+            await this.page.locator('label[for="shippingAddress.shouldSaveAddress"]').click();
+
+        await this.page.locator('#checkout-shipping-options').waitFor({ state: 'visible' });
+
+        const firstShippingMethod = this.page
+            .locator('#checkout-shipping-options .shippingOptions-container li')
+            .nth(0);
+
+        await firstShippingMethod.click();
+        await this.page.locator('#checkout-shipping-continue').click();
+    }
+
+    async completeBillingAddressForm(address: Address): Promise<void> {
+        await this.page
+            .locator('[data-test="billing-address-heading"]')
+            .waitFor({ state: 'visible' });
+
+        await this.fillAddressForm({ formId: '#checkoutBillingAddress', address });
+
+        await this.page.locator('#checkout-billing-continue').click();
+    }
+
+    async fillNewAddressPopup(address: Address, addressAppendText: string): Promise<void> {
+        await this.page.locator('.dropdownMenu').waitFor({ state: 'visible' });
+        await this.page.locator('.dropdownMenu li').nth(0).click();
+        await this.page
+            .locator('.ReactModalPortal .modal.optimizedCheckout-contentPrimary')
+            .waitFor({ state: 'visible' });
+        await this.fillAddressForm({
+            formId: '[data-test="modal-body"] .checkout-address',
+            address,
+            addressAppendText,
+        });
+        await this.page.locator('#checkout-save-address').click();
+    }
+
+    async completeMultiShippingAddressForm(address: Address): Promise<void> {
```

Comment:
I think a developer can hardly figure out what the cart and account should look like to use this method.
Also, there are too many ways to complete a multi shipping form.
Perhaps we should put playwright actions directly in the test.